### ISSUE TITLE
mdOption repeating n^2 times

### ISSUE THREAD

**Issue by DavidSouther** [angular/material#1753]
I'm truly flabbergasted whether this is a bug in Material or in my code.

![screen shot 2015-03-02 at 9 46 13 pm](https://cloud.githubusercontent.com/assets/498712/6455692/9774d5e2-c125-11e4-9c8b-aed12699e014.png)

In this screenshot, you can see two "Glass" and the (default selected) "Steel" 4 times. When scrolling, each item is repeated a number of times equal to the number of items in the list.

There are 7 items in the data array, which is built with `Object.keys` from a keyed data store. If I click on the first of any group of items, I get the "first" item in the array; second of a group gets the second from the array.

Data Source (Apologies for the coffeescript and jade, if it's a problem I'll repost in JS/HTML):

```
module.value 'MaterialList',
  "Concrete":
    density: 23e3
    elasticity: 20e9
    stress: 70e6
    color: 'bisque'
  "Granite":
    density: 27e3
    elasticity: 80e9
    stress: 150e6
    color: 'slategrey'
  "Plastic":
    density: 27e3
    elasticity: 2e9
    stress: 50e6
    color: 'honeydew'
```

```
$scope.materials = Object.keys MaterialList
```

```
        md-select(
          ng-model="materialName"
          placeholder="Material"
          multiple=false
        )
          md-option(
            ng-repeat="material in materials"
            ng-value="material"
          ) {{ material }}
```

**Comment by gkalpak** [comment]
There was indeed such a bug, but I think it was fixed recently.
Have you tried with version currently in master ?

**Comment by Frank3K** [comment]
This was fixed in 9e0ca44.

**Comment by DavidSouther** [comment]
Obviously a dupe of #1676 not sure how I missed that. (And I usually stick with point releases.)
